ci-operator/step-registry/openshift/image-based/upgrade/metal/config/openshift-image-based-upgrade-metal-config-commands.sh

```diff
@@ -0,0 +1,177 @@
+#!/bin/bash
+
+set -x
+set -o nounset
+set -o errexit
+set -o pipefail
+export PS4='+ $(date "+%T.%N") \011'
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-privatekey")
+
+instance_ip=$(cat ${SHARED_DIR}/public_address)
+host=$(cat ${SHARED_DIR}/ssh_user)
+
+ssh_host_ip="$host@$instance_ip"
+
+if ! test -f "${SHARED_DIR}/remote_workdir"; then
+  workdir="/home/${host}/workdir-seed-$(date +%Y%m%d)"
+
+  echo "${workdir}" >> "${SHARED_DIR}/remote_workdir"
+fi
+
+remote_workdir=$(cat "${SHARED_DIR}/remote_workdir")
+
+ssh "${SSHOPTS[@]}" "${ssh_host_ip}" "mkdir -p ${remote_workdir}"
+
+cat <<EOF > ${SHARED_DIR}/install.sh
+#!/bin/bash
+set -xeuo pipefail
+
+sudo subscription-manager config --rhsm.manage_repos=1 --rhsmcertd.disable=redhat-access-insights
+
+if ! sudo subscription-manager status >&/dev/null; then
+    sudo subscription-manager register \
+        --org="\$(cat /tmp/subscription-manager-org)" \
+        --activationkey="\$(cat /tmp/subscription-manager-act-key)"
+fi
+
+sudo subscription-manager repos \
+--enable "rhel-9-for-$(uname -m)-appstream-rpms" \
+--enable "rhel-9-for-$(uname -m)-baseos-rpms" \
+--enable "rhocp-4.14-for-rhel-9-$(uname -m)-rpms"
+
+cd ${remote_workdir}
+
+sudo dnf -y copr enable nmstate/nmstate-git
+sudo dnf -y clean all
+sudo dnf -y update
+sudo dnf -y install nmstate virt-install virt-manager libvirt-nss openshift-clients cockpit-machines golang jq
+
+sudo systemctl start libvirtd
+sudo systemctl enable libvirtd
+
+sudo usermod -a -G libvirt ec2-user
+
+git clone https://github.com/rh-ecosystem-edge/ib-orchestrate-vm.git
```

Comment:
@javipolo note that the repo is being used in the CI

Reply:
A while back I had talked with Javi about using the tools he created in CI so he should know